### ISSUE TITLE
Upload of mp3 and ogg files fail on debian jessie

### ISSUE THREAD

**Issue by Anubys** [LibreTime/libretime#352]
Hi
On a fresh debian , the install file from git works well but the upload of mp3 and  ogg failed
From /var/log/airtime/airtime_analyzer.log :
2017-11-13 10:55:02,724 [message_listener] [INFO ]   - Received '{"tmp_file_path":"\/srv\/airtime\/stor\/organize\/Sample_of__Another_Day_in_Paradise_.ogg","storage_backend":"file","import_directory":"\/srv\/airtime\/stor\/\/imported\/1","original_filename":"Sample_of__Another_Day_in_Paradise_.ogg","callback_url":"http:
2017-11-13 10:55:04,812 [replaygain_analyzer] [WARNI]  ['replaygain', '-d', u'/srv/airtime/stor/organize/Sample_of__Another_Day_in_Paradise_.ogg']  1
2017-11-13 10:55:05,123 [playability_analyzer] [WARNI]  Command '['liquidsoap', '-v', '-c', 'output.dummy(audio_to_stereo(single(argv(1))))', '--', u'/srv/airtime/stor/organize/Sample_of__Another_Day_in_Paradise_.ogg']' returned non-zero exit status 1
2017-11-13 10:55:05,123 [analyzer_pipeline] [ERROR]
Traceback (most recent call last):

 **File "/usr/local/lib/python2.7/dist-packages/airtime_analyzer-0.1-py2.7.egg/airtime_analyzer/analyzer_pipeline.py", line 73, in run_analysis
    metadata = PlayabilityAnalyzer.analyze(audio_file_path, metadata)
  File "/usr/local/lib/python2.7/dist-packages/airtime_analyzer-0.1-py2.7.egg/airtime_analyzer/playability_analyzer.py", line 30, in analyze**
    raise UnplayableFileError()

My silan version is:
silan version 0.3.3 

Thanks 

MAINTAINER EDIT - [won't fix reasoning](https://github.com/LibreTime/libretime/issues/352#issuecomment-506953920)

**Comment by Robbt** [comment]
Does this happen to all mp3 and ogg or just specific ones ?

**Comment by Anubys** [comment]
All mp3 files and ogg :( 
Even this  https://commons.wikimedia.org/wiki/File:Median_test.ogg 

Thnx

**Comment by Robbt** [comment]
Ok, it may be an issue with your liquidsoap then. What version of liquidsoap is reported when you type liquidsoap -v

There was a user experiencing a similar problem reported on bug #340 but I'm not sure that it was ever fully resolved. That issue contains some of the liquidsoap commands parsed so that they can be tested from the command line.

We can get to the bottom of this. I'll test it on my debian set-up to see if I can replicate it.

**Comment by hairmare** [comment]
Can you try playing the file through liquidsoap directly? Maybe it outputs more info on the error.

```
liquidsoap --verbose --debug -c 'output.dummy(audio_to_stereo(single("/path/to/file.ogg")))'
```

I did a quick test with the file from wikimedia commons and there does seem to be something fishing going on:

```
notification: This doesn't look like a Speex file
notification: This doesn't look like a Speex file
```

Also, when I downloaded it, it got saved with an `.ogx` extension rather than `.ogg`... This seems to indicate that it's an Ogg Vorbis Multiplexed Media File file but I haven't encountered one of those before.

**Comment by Anubys** [comment]
Hi

liquidsoap test dont show  errors 

 liquidsoap --verbose --debug -c 'output.dummy(audio_to_stereo(single("unknown-Median_test-48kbps.ogg")))'
liquidsoap --verbose --debug -c 'output.dummy(audio_to_stereo(single("unknown-Sample_of__Another_Day_in_Paradise_-64kbps.ogg")))'

If run `liquidsoap -v `the console show the next message:
**No output defined, nothing to do.**

Thanks

**Comment by Robbt** [comment]
Ok¸ I have confirmed this. It appears that the version of liquidsoap included with debian is broken for the importation of mp3 files. I'm not sure what we can do about it, but I get segmentation faults when trying to do any imports of mp3s. Ubuntu works out of the box, so we either need to only support Ubuntu or figure out a way to automate the installation of a newer version of liquidsoap for our debian users from the install.

**Comment by hairmare** [comment]
Since this seems to be the issue that we're keeping open I'd like to add that it seems the mp3 upload issue exists depending on the underlying system and also affect Debian Stretch installs with a os distributed liquidsoap.

Since upgrading to liquidsoap works the workaround to this issue is to either us an opam install of liquidsoap 1.3.x, switch to ubuntu or centos (either with or without intalling 1.3.x) or help out with native [Debian packaging](https://github.com/LibreTime/libretime-debian-packaging).

I think I also saw it mentioned somewhere that the underlying host system makes a difference, ie. it works on vagrant but not on some VPN services.

**Comment by Robbt** [comment]
This needs testing under Debian 9 aka Stretch to confirm otherwise we can't support Debian until Debian 10 is released or this is fixed.

**Comment by paddatrapper** [comment]
I am happy to test this on Stretch and Buster (10) over the next couple of days

**Comment by paddatrapper** [comment]
Tested under stretch and had the same error occur:

```
2018-11-30 19:42:25,529 [message_listener] [INFO ]   - Received '{"tmp_file_path":"\/srv\/airtime\/stor\/organize\/Breakeven.mp3","storage_backend":"file","import_directory":"\/srv\/airtime\/stor\/\/imported\/1","original_filename":"Breakeven.mp3","callback_url":"http:\/\/lt-stretch.paddatrapper.com:443\/\/rest\/media\/1","api_key":"RBZ6ALPFSW5L0DP26R5Y","file_prefix":""}' on routing_key ''
2018-11-30 19:42:25,717 [cuepoint_analyzer] [WARNI]  Failed to run: silan - No such file or directory. Do you have silan installed?
2018-11-30 19:42:30,325 [playability_analyzer] [WARNI]  Command '['liquidsoap', '-v', '-c', 'output.dummy(audio_to_stereo(single(argv(1))))', '--', u'/srv/airtime/stor/organize/Breakeven.mp3']' returned non-zero exit status -11
2018-11-30 19:42:30,327 [analyzer_pipeline] [ERROR]  
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/airtime_analyzer-0.1-py2.7.egg/airtime_analyzer/analyzer_pipeline.py", line 69, in run_analysis
    metadata = PlayabilityAnalyzer.analyze(audio_file_path, metadata)
  File "/usr/local/lib/python2.7/dist-packages/airtime_analyzer-0.1-py2.7.egg/airtime_analyzer/playability_analyzer.py", line 30, in analyze
    raise UnplayableFileError()
UnplayableFileError
```

Running liquidsoap directly results in:

```
$ liquidsoap -v -c 'output.dummy(audio_to_stereo(single(argv(1))))' -- /srv/airtime/stor/organize/Breakeven.mp3
Segmentation Fault
```

**Comment by paddatrapper** [comment]
I am going to try get a fix to 1.1.1 based on Ubuntu's changes uploaded to Debian stable ASAP (busy building and testing it at the moment)

**Comment by Robbt** [comment]
Hey @paddatrapper did you have any luck with this or should we just consider only supporting Buster aka Debian 10 once it is out and discouraging use of previous Debian versions due to this bug.

**Comment by paddatrapper** [comment]
I'm still waiting for approval. The request bug can be found [here](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=916650)

**Comment by paddatrapper** [comment]
I suggest we close this as won't fix - Buster is about to be released with a new version of Liquidsoap and it seems that my request for a fix upload to Stretch is not going to be accepted. We should just support Debian Buster and higher

**Comment by Robbt** [comment]
Sounds good. Yeah focusing on Buster support is a good idea.

**Comment by paddatrapper** [comment]
And it was finally approved! I uploaded a fixed version of liquidsoap 1.1.1 to Debian Stretch today. It should be on the mirrors soon.
